src/main/java/com/hcl/appscan/jenkins/plugin/scanModes/ScanModeFactory.java

```diff
@@ -0,0 +1,17 @@
+package com.hcl.appscan.jenkins.plugin.scanModes;
+
+public class ScanModeFactory {
+
+    public static ScanMode getScanTypeImplementation(String scanType, String loginType, String userName, String password, String trafficFile, String exploreData) {
+        switch (scanType) {
```

Comment:
Is there a reason we're using String representations of numbers here? Can we use an enum instead?

Reply:
Sure, now I am using enums for the comparison.